### PR TITLE
Enable Dependabot for the repository

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Because why not? We will not be forcing Jenkins core and plugins, but developer tools and Maven plugins can be updated automagically

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

File: .dependabot/config.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+version: 1
+update_configs:
+- package_manager: "java:maven"
+  directory: "/"
+  update_schedule: "weekly"
+  ignored_updates:
+    - match:
+        dependency_name: "org.jenkins-ci.main:jenkins-core"
+    - match:
+        dependency_name: "org.jenkins-ci.plugins*"
```
Comment (casz) [inline .dependabot/config.yml]: wouldn't this block POM? updates

Comment (oleg-nenashev) [reply .dependabot/config.yml]: Right, it will. Need a smart regexp

Comment (oleg-nenashev) [reply .dependabot/config.yml]: Not trivial to implement unfortunately, I will think how to do it later

Comment (casz) [reply .dependabot/config.yml]: No worries let's get going 😓 I think let's get vault dep moved out before it starts suggesting to update vault-java-driver

**Comment by oleg-nenashev** [comment]
OK, let's see how it flies

**Pull request merged**